docs/release-notes.rst

```diff
@@ -19,4 +19,179 @@
 
 *************
 Release Notes
-*************
\ No newline at end of file
+*************
+
+.. contents:: On this page
+   :depth: 4
+   :local:
+   :backlinks: none
+
+========================
+Changes and New Features
+========================
+
+.. list-table::
+   :header-rows: 1
+   :widths: 10, 75
```

Comment:
Do we need the widths? I guess it can auto adjust based on the text

Reply:
Here how it looks without width
![image](https://github.com/Mellanox/network-operator-docs/assets/12015519/a4adc525-87d0-43a6-875b-066148012efb)